turbopack/crates/turbopack-core/src/data_uri_source.rs

```diff
@@ -0,0 +1,79 @@
+use std::hash::BuildHasher;
+
+use anyhow::{bail, Result};
+use rustc_hash::FxBuildHasher;
+use turbo_rcstr::RcStr;
+use turbo_tasks::{ResolvedVc, Vc};
+use turbo_tasks_fs::{rope::Rope, File, FileContent, FileSystemPath};
+use turbo_tasks_hash::encode_hex;
+
+use crate::{
+    asset::{Asset, AssetContent},
+    ident::AssetIdent,
+    source::Source,
+};
+
+/// The raw [Source]. It represents raw content from a path without any
+/// references to other [Source]s.
+#[turbo_tasks::value]
+pub struct DataUriSource {
+    media_type: RcStr,
+    encoding: RcStr,
+    data: ResolvedVc<RcStr>,
+    lookup_path: ResolvedVc<FileSystemPath>,
+}
+
+#[turbo_tasks::value_impl]
+impl DataUriSource {
+    #[turbo_tasks::function]
+    pub fn new(
+        media_type: RcStr,
+        encoding: RcStr,
+        data: ResolvedVc<RcStr>,
+        lookup_path: ResolvedVc<FileSystemPath>,
+    ) -> Vc<Self> {
+        Self::cell(DataUriSource {
+            media_type,
+            encoding,
+            data,
+            lookup_path,
+        })
+    }
+}
+
+#[turbo_tasks::value_impl]
+impl Source for DataUriSource {
+    #[turbo_tasks::function]
+    async fn ident(&self) -> Result<Vc<AssetIdent>> {
+        let content_type = self.media_type.split(";").next().unwrap().into();
+        let filename = format!(
+            "data-{}",
```

Comment:
```suggestion
            "data:{}",
```

I would use a colon to make it look more like a data url